tt_metal/impl/program/program.cpp

```diff
@@ -610,14 +610,14 @@ void Program::compile( Device * device )
     }
 
     for (auto & f : events)
-        f.wait();
+        f.get();
```

Comment:
The problem I ran into here was that `future::wait()` discards any exceptions thrown by the event it's waiting for, meaning that a silent failure in a previous test that used the same kernel as my new test was causing only the new test to fail. I've confirmed `future::get()` does throw in the correct test.

Reply:
Is this the source of that CI map error?

Reply:
This + the fix in `runtime_args.cpp` were the cause of that CI map error

Reply:
nice catch!